src/dune_rules/odoc.ml

```diff
@@ -443,20 +442,19 @@ let libs_of_pkg ctx ~pkg =
   (* Filter out all implementations of virtual libraries *)
   List.filter_map entries ~f:(fun (entry : Scope.DB.Lib_entry.t) ->
       match entry with
+      | Deprecated_library_name _ -> None
       | Library lib ->
         let is_impl =
           Lib.Local.to_lib lib |> Lib.info |> Lib_info.implements
           |> Option.is_some
         in
-        Option.some_if (not is_impl) lib
-      | Deprecated_library_name _ -> None)
+        Option.some_if (not is_impl) lib)
 
 let load_all_odoc_rules_pkg sctx ~pkg =
   let* pkg_libs = libs_of_pkg sctx ~pkg in
   let+ () =
-    Memo.parallel_iter
-      (Pkg pkg :: List.map pkg_libs ~f:(fun lib -> Lib lib))
-      ~f:(fun _ -> Memo.return ())
+    Pkg pkg :: List.map pkg_libs ~f:(fun lib -> Lib lib)
+    |> Memo.parallel_iter ~f:(fun _ -> Memo.return ())
```

Comment:
`Memo.all_concurrently`?

Reply:
Doesn't help. We would need to ignore the list of units in the end.

Reply:
ah I was confused. Fiber has an `all_concurrently_unit` that doesn't allocate a return list.

Reply:
In fact Memo already has it since it `include`s `Fiber`. All you need to do is add
```
val all_concurrently_unit : unit t list -> unit t
```
after `all_concurrently` after `memo.mli`.